src/librustc_codegen_llvm/back/linker.rs

```diff
@@ -978,9 +980,18 @@ impl Linker for WasmLd {
     }
 
     fn gc_sections(&mut self, _keep_metadata: bool) {
+        self.cmd.arg("--gc-sections");
     }
 
     fn optimize(&mut self) {
+        self.cmd.arg(match self.sess.opts.optimize {
+            OptLevel::No => "-O0",
+            OptLevel::Less => "-O1",
+            OptLevel::Default => "-O2",
+            OptLevel::Aggressive => "-O3",
+            OptLevel::Size => "-O2",
```

Comment:
If this is intentional, it'd be helpful to have a comment here to explain why `-O2` is used for the size opt levels (as opposed to `-Os` and `-Oz`).

Reply:
Sure thing!